ipaserver/plugins/ldap2.py

```diff
@@ -268,6 +268,17 @@ def has_upg(self):
         org_filter = upg_entries[0].single_value['originfilter']
         return '(objectclass=disable)' not in org_filter
 
+    def __set_get_effective_rights_control(self):
+        bind_dn = self.conn.whoami_s()[4:]
+        sctrl = [
+            GetEffectiveRightsControl(
+                True, "dn: {0}".format(bind_dn).encode('utf-8'))
+        ]
+        self.conn.set_option(_ldap.OPT_SERVER_CONTROLS, sctrl)
+
+    def __unset_get_effective_rights_control(self):
+        self.conn.set_option(_ldap.OPT_SERVER_CONTROLS, [])
```

Comment:
This is a nasty hack around bad API design in freeIPA's LDAP code. python-ldap supports per-call server and client controls, https://www.python-ldap.org/en/latest/reference/ldap.html#arguments-for-ldapv3-controls . I'd rather have the controls in the call rather than messing with global settings.

Reply:
Good idea.  Will do.